### PR TITLE
Adding combined accuracy and runtime tables

### LINKED ISSUE #137: Compare run time and accuracy in one place with colours
Feedback from Tyrone:

He would like there to be a way for the user to have an indication of the accuracy and run time in one place. A suggestion was to use the outline colours of the runtime cells on an accuracy table and visa.

Further implement this to be an user option, i.e. something which the user can turn off and on, implemented after #167 is completed.

### PR DESCRIPTION
**Pull request by bb511**
#### Description of Work

Fixes #137 

FitBenchmarking now outputs an additional table that has combined accuracy and runtime information.

#### Testing Instructions

1. Run `example_runScripts.py` in the mantidpython console.
2. Check the results folder and the combined table in question.

### CODE REVIEW

**Comment by Anders-Markvardsen** [comment]
@bb511 is there a reason why results are saved according to "If left as None, they will be saved in a "results" folder in the working dir" and not it the directory where the script is?

**Comment by Anders-Markvardsen** [comment]
Observe when running the example_script with 'scipy' and 'nist' no combined_table outputted

**Comment by bb511** [comment]
> @bb511 is there a reason why results are saved according to "If left as None, they will be saved in a "results" folder in the working dir" and not it the directory where the script is?

No particular reason, I thought it would be easier for the user that way.

**Comment by bb511** [comment]
> Observe when running the example_script with 'scipy' and 'nist' no combined_table outputted

I will look into that ASAP.

**Comment by Anders-Markvardsen** [comment]
Is there a result that you have not added a unit test for this new functionality, thanks

**Comment by Anders-Markvardsen** [comment]
@wathen I am wondering if it is better to wait with completing this issue until after the structural changes you suggested last Friday has been implemented? We can discuss this this Friday

**Comment by wathen** [comment]
> @wathen I am wondering if it is better to wait with completing this issue until after the structural changes you suggested last Friday has been implemented? We can discuss this this Friday

Yea this makes sense

**Comment by bb511** [comment]
@Anders-Markvardsen sorry I could not reproduce the bug you mentioned. When I run scipy and nist, I produce a combined table for each level of difficulty.

File: fitbenchmarking/resproc/misc.py [hunk 1]
```diff
@@ -97,3 +98,21 @@ def build_items_links(comparison_type, comp_dim, using_errors):
         items_link = ''
 
     return items_link
+
+def find_ranking_strings(rst_table):
+    """
+    Gets the intervals between which a table has ':ranking-number:' and
```
Comment (AnthonyLim23) [inline fitbenchmarking/resproc/misc.py]: the "which a" makes it a bit difficult to understand the comment

File: fitbenchmarking/resproc/misc.py [hunk 2]
```diff
@@ -97,3 +98,21 @@ def build_items_links(comparison_type, comp_dim, using_errors):
...
+
+    @param rst_table :: the restructured text table containing the results.
+
+    @returns :: array of tuples containing the starting and ending index
+                where every ':ranking-number:' is.
+    """
+
+    expression = re.compile('(:ranking)(.*?)(:)')
+    iterator = expression.finditer(rst_table)
+    matches = []
```
Comment (AnthonyLim23) [inline fitbenchmarking/resproc/misc.py]: what would happen if two different rows had the same ranking number?

File: fitbenchmarking/resproc/tests/test_misc.py [hunk 3]
```diff
@@ -106,6 +107,15 @@ def test_buildItemsLinks_return_empty_itemsLinks_invalid_comparison(self):
 
         self.assertEqual(items_link_expected, items_link)
 
+    def test_findRankingStrings_get_ranking_strings(self):
+
+        rst_table = "something something :ranking-9 example: something"
+
+        matches = find_ranking_strings(rst_table)
+        expected_matches = [(20,39)]
```
Comment (AnthonyLim23) [inline fitbenchmarking/resproc/tests/test_misc.py]: Its not clear to my why the answer should be 20,39

File: fitbenchmarking/results_output.py [hunk 4]
```diff
@@ -70,6 +71,10 @@ def save_results_tables(software, results_per_test, group_name,
     runtime_tbl = create_runtime_tbl(minimizers, linked_problems, norm_runtimes,
                                      use_errors, color_scale)
 
+    combined_table = create_combined_table(acc_tbl, runtime_tbl)
+
+    save_tables(tables_dir, combined_table, use_errors, group_name,
+                "combined_table")
```
Comment (AnthonyLim23) [inline fitbenchmarking/results_output.py]: do we want to save the combined and individual tables?

**Comment by Anders-Markvardsen** [comment]
This PR will be re-considered after issue #154 and #155 complete

**Comment by Anders-Markvardsen** [comment]
The code in this PR first need to be merged with recent larger structural changes to the code, which is possibly best done after #165

**Comment by bb511** [comment]
@AtomAnu If you have some time, would you be able to look over this PR and see if it could be merged?

**Comment by AtomAnu** [comment]
@bb511 Sure, I'll look into it.

**Comment by AndrewLister-STFC** [comment]
Looking at this as it seems to have been waiting since Feb.

The issue states that it should show an indication of the other tables value (i.e. a coloured outline for each cell.) This PR creates a new 3rd table with the accuracy results and timing colours, whereas I was under the impression that the request was to augment the existing tables?

Personally I find this creates tables which can be confusing, as below. To me this looks like we have rounding issues in the number displayed, when it's just showing the runtimes colour:
image

**Comment by tyronerees** [comment]
Discussed in meeting -- one table with accuracy and runtime in the same table, with two rows per problem, would be easier to read

**Pull request closed without merging**